lib/Elevate/Components/MySQL.pm

```diff
@@ -360,6 +362,41 @@ sub _blocker_mysql_upgrade_in_progress ($self) {
     return 0;
 }
 
+sub _blocker_mysql_database_corrupted ($self) {
+
+    # We cannot run mysqlcheck if mysql is not running
+    if ( !Cpanel::MysqlUtils::Running::is_mysql_running() ) {
+
+        if ( Cpanel::Services::Enabled::is_enabled('mysql') ) {
+            my $db_version = Elevate::Database::get_local_database_version();
+            my $db_type    = Elevate::Database::get_database_type_name_from_version($db_version);
+
+            WARN("$db_type is not running; this prevents us from verifying the integrity of the databases.");
+        }
+
+        return 0;
+    }
+
+    # Perform a medium check on all databases and only output errors
+    my $mysqlcheck_path = Cpanel::Binaries::path('mysqlcheck');
+    my $output          = $self->ssystem_capture_output( $mysqlcheck_path, '-c', '-m', '-A', '--silent' );
+
+    # mysqlcheck doesn't return an error code
+    # We check for lines that actually begin with "Error" (or "error")
+    # because we don't want to block of only warnings are found
+    if ( scalar grep { /^[eE]rror/ } @{ $output->{stdout} } ) {
```

Comment:
CONSIDER: Ignore case in the regex here so that we catch ERROR here too.  Not sure if this is necessary.

Reply:
Yeah, I think that is a better approach.